### PR TITLE
FOGL-1079 unit tests added for services.core.connect

### PR DESCRIPTION
**Pull request by ashish-jabble**
##### Coverage for /home/foglamp/Development/FogLAMP/python/foglamp/services/core/connect.py : 100% 
`17 statements   17 run 0 missing 0 excluded`

**Tests O/P**
```
collected 3 items                                                                                                                             

services/core/test_connect.py::TestConnect::test_get_storage PASSED
services/core/test_connect.py::TestConnect::test_get_storage_exception PASSED
services/core/test_connect.py::TestConnect::test_get_storage_exception_when_non_foglamp_storage PASSED

=== 3 passed in 0.03 seconds ===
```
**In suite (No regression "/tests/unit/python/foglamp")**
```
collected 180 items                                                                                                                           

common/test_audit_logger.py ..........                                                                                                  [  5%]
common/test_statistics.py ................                                                                                              [ 14%]
common/storage_client/test_payload_builder.py ...................................................s...........                           [ 49%]
common/storage_client/test_sc_exceptions.py .........                                                                                   [ 54%]
common/storage_client/test_utils.py ..................                                                                                  [ 64%]
plugins/common/test_plugins_common_utils.py ....                                                                                        [ 66%]
plugins/north/common/test_common.py .                                                                                                   [ 67%]
plugins/south/test_http_south.py ...                                                                                                    [ 68%]
services/common/test_avahi.py ......                                                                                                    [ 72%]
services/common/microservice_management/test_instance.py ..........                                                                     [ 77%]
services/core/test_connect.py ...                                                                                                       [ 79%]
services/core/test_main.py .                                                                                                            [ 80%]
services/core/api/test_audit.py .............                                                                                           [ 87%]
services/core/api/test_common_ping.py ..                                                                                                [ 88%]
services/core/service_registry/test_exceptions.py .....                                                                                 [ 91%]
tasks/purge/test_purge.py ................                                                                                              [100%]

== 179 passed, 1 skipped in 2.05 seconds ===
```

### CODE REVIEW

File: tests/unit/python/foglamp/services/core/test_connect.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            assert isinstance(storage_client, StorageClient)
+
+    def test_get_storage_exception(self):
+        service_reg = MagicMock(spec=Service)
+        with patch.object(service_reg, 'get', side_effect=Exception()):
+            with pytest.raises(DoesNotExist) as excinfo:
+                with patch.object(connect._logger, 'exception') as logger_exception:
+                    connect.get_storage()
+                assert str(excinfo).endswith('DoesNotExist')
+                logger_exception.assert_called_once_with()
```
Comment (praveen-garg) [inline tests/unit/python/foglamp/services/core/test_connect.py]: there is nothing `_with()` ? there must be something to log

File: tests/unit/python/foglamp/services/core/test_connect.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+from unittest.mock import MagicMock, patch
+import pytest
+
+from foglamp.services.core.service_registry.service_registry import ServiceRegistry as Service
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/test_connect.py]: Why use alias? Essentially all imports in unit test should comply with the original code.

File: tests/unit/python/foglamp/services/core/test_connect.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+@pytest.allure.story("services", "core")
+class TestConnect:
+    """ Storage connection"""
+    
+    def setup_method(self):
+        Service._registry = []
+
+    def teardown_method(self):
+        Service._registry = []
+
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/test_connect.py]: Not sure about the use of setup and teardown. Can we get rid of classic xunit-style setup?

File: tests/unit/python/foglamp/services/core/test_connect.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    def test_get_storage(self):
+        service_reg = MagicMock(spec=Service)
+        service_idx = Service.register("FogLAMP Storage", "Storage", "127.0.0.1", 37449, 37843)
+        with patch.object(service_reg, 'get', return_value=service_idx):
+            storage_client = connect.get_storage()
+            assert isinstance(storage_client, StorageClient)
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/test_connect.py]: More assertions, like ServiceRegistry.get called once with params name="FogLAMP Storage" required?

File: tests/unit/python/foglamp/services/core/test_connect.py (hunk 3 above)
Comment (praveen-garg) [reply tests/unit/python/foglamp/services/core/test_connect.py]: any specific reason? we should start using them at each level as needed to keep state clean

**Pull request merged**